packages/components/src/internal/components/settings/DeleteProjectModal.tsx

```diff
@@ -0,0 +1,208 @@
+import { Summary } from '../security/APIWrapper';
+
+interface CommentAreaProps {
+    comment: string;
+    onChangeComment: ChangeEventHandler<HTMLTextAreaElement>;
+}
+export const CommentArea: FC<CommentAreaProps> = memo(({ comment, onChangeComment }) => {
+    return (
+        <div className="delete-project-modal__comment">
+            <p>
+                <b>Deletion cannot be undone.</b> Do you want to proceed?{' '}
+            </p>
+            <p>
+                <b>Reason for deleting</b>
+            </p>
+            <textarea
+                className="form-control"
+                placeholder="Enter comments (optional)"
+                value={comment}
+                rows={5}
+                cols={50}
+                onChange={onChangeComment}
+            />
+        </div>
+    );
+});
+
+export const BodyEmpty: FC<CommentAreaProps> = memo(({ comment, onChangeComment }) => {
+    return (
+        <Modal.Body>
+            <div className="delete-project-modal__text">
+                <p>This project will be permanently deleted. It contains no data.</p>
+            </div>
+
+            <CommentArea comment={comment} onChangeComment={onChangeComment} />
+        </Modal.Body>
+    );
+});
+
+interface BodyProps {
+    comment: string;
+    onChangeComment: ChangeEventHandler<HTMLTextAreaElement>;
+    summaries: Summary[];
+}
+export const Body: FC<BodyProps> = memo(({ summaries, comment, onChangeComment }) => {
+    return (
+        <Modal.Body>
+            <div className="delete-project-modal__text">
+                <p>This project and all of its data will be permanently deleted.</p>
+                <p>
+                    Before deleting this project, ensure there are no references to data (samples, sources or registry,
+                    assay data, etc.) in other projects.
+                </p>
+            </div>
+
+            <b> Project Data </b>
+
+            {summaries ? (
+                <table className="table table-responsive table-condensed delete-project-modal__table ">
+                    <tbody>
+                        {summaries.map(s => (
+                            <tr key={s.count + s.noun}>
+                                <td>{s.noun}</td>
+                                <td>{s.count}</td>
+                            </tr>
+                        ))}
+                    </tbody>
+                </table>
+            ) : (
+                <LoadingSpinner wrapperClassName="delete-project-modal__spinner" />
+            )}
+
+            <CommentArea comment={comment} onChangeComment={onChangeComment} />
+        </Modal.Body>
+    );
+});
+
+interface BodyDeletingProps {
+    totalCountFromSummaries: number;
+}
+export const BodyDeleting: FC<BodyDeletingProps> = memo(({ totalCountFromSummaries }) => {
+    const [toggle, seToggle] = React.useState<boolean>(false);
+
+    React.useEffect(() => {
+        setTimeout(() => seToggle(true), 1);
```

Comment:
Why not just toggle=true on the component declaration below? If the implementation is left this way then please consider leaving a comment as to why a timeout is necessary.